packages/editor/src/extensions/attachment/component.tsx

```diff
@@ -84,30 +119,85 @@ export function AttachmentComponent(props: ReactNodeViewProps<FileAttachment>) {
       >
         {progress ? `${progress}%` : formatBytes(size)}
       </Text>
-      <DesktopOnly>
-        {selected && !isDragging && (
-          <ToolbarGroup
-            editor={editor}
-            groupId="attachmentTools"
-            tools={
-              editor.isEditable
-                ? [
-                    "removeAttachment",
-                    "downloadAttachment",
-                    "previewAttachment"
-                  ]
-                : ["downloadAttachment", "previewAttachment"]
+    </Box>
+  );
+
+  // Common toolbar component
+  const renderToolbar = () => (
+    <DesktopOnly>
+      {selected && !isDragging && (
+        <ToolbarGroup
+          editor={editor}
+          groupId="attachmentTools"
+          tools={getToolbarTools()}
+          sx={{
+            boxShadow: "menu",
+            borderRadius: "default",
+            bg: "background",
+            position: "absolute",
+            top: -35
+          }}
+        />
+      )}
+    </DesktopOnly>
+  );
+
+  // Audio player section for audio files
+  const renderAudioPlayer = () => (
+    <Box
+      sx={{
+        width: "100%",
+        "& audio": {
+          width: "100%",
+          height: "32px"
+        }
+      }}
+    >
+      <audio controls preload="metadata" src={audioSrc} />
+    </Box>
+  );
+
+  return (
+    <Box
+      ref={elementRef}
+      as="span"
+      contentEditable={false}
+      variant={"body"}
+      sx={{
+        display: "inline-flex",
+        position: "relative",
+        userSelect: "none",
+        backgroundColor: "var(--background-secondary)",
+        m: 1,
+        borderRadius: "default",
+        border: "1px solid var(--border)",
+        borderColor: selected ? "accent" : "border",
+        ":hover": {
+          bg: "hover"
+        },
+        // Audio-specific overrides
+        ...(isAudioFile && audioSrc
+          ? {
+              flexDirection: "column",
+              p: 2,
+              width: "50%",
             }
-            sx={{
-              boxShadow: "menu",
-              borderRadius: "default",
-              bg: "background",
-              position: "absolute",
-              top: -35
-            }}
-          />
-        )}
-      </DesktopOnly>
+          : {
+              justifyContent: "center",
+              alignItems: "center",
+              px: 1,
+              cursor: "pointer",
+              maxWidth: 250
+            })
+      }}
+      title={!isAudioFile || !audioSrc ? filename : undefined}
+      onDragStart={() => setIsDragging(true)}
+      onDragEnd={() => setIsDragging(false)}
+      data-drag-handle
+    >
+      {renderFileInfo()}
+      {isAudioFile && audioSrc && renderAudioPlayer()}
+      {renderToolbar()}
```

Comment:
Don't think we need to separate out into functions when we can write JSX directly here